core/primitives-core/src/config.rs

```diff
@@ -387,6 +387,15 @@ pub struct ExtCostsConfig {
     /// Cost of calling `validator_total_stake`.
     pub validator_total_stake_base: Gas,
 
+    // Workaround to keep JSON serialization backwards-compatible
+    // <https://github.com/near/nearcore/pull/6587#discussion_r876113324>.
+    //
+    // Remove once #5516 is fixed.
+    #[serde(default, rename = "contract_compile_base")]
+    pub _unused1: Gas,
```

Comment:
How about we do this instead?

```suggestion
    #[serde(default)]
    pub contract_compile_base: Option<Gas>,
```

Reply:
I rather like that `_unused1` very clearly signals that you shouldn't use that. The `Option` would also get serialized as `null`, which I think might break somebody?

Reply:
Why don’t we set the values to be the same as contract_loading_base and contract_loading_bytes respectively? To me it seems like a proper way to rename a field is to keep the old one for a couple releases to give people time to migrate.

Reply:
We can do that, but that requires writing custom code which might contain bugs. The current solution is significantly simpler than that. 

My understanding that for this code we want to prioritize simplicity over compatibility. I *think* the problem here is just that the Rust code fails with deserialization error, rather than that someone actually looks at the old field.